src/components/CompletingParty/CompletingParty.vue

```diff
@@ -0,0 +1,287 @@
+<template>
+  <section id="completing-party" class="section-container" :class="{'invalid-section': invalidCompletingParty }">
+    <!-- Add/Edit Block -->
+    <template v-if="enableAddEdit">
+      <v-row no-gutters>
+        <v-col cols="12" sm="3">
+          <label class="add-person-header">
+            <span :class="{ 'error-text': invalidCompletingParty }">Add Person</span>
+          </label>
+        </v-col>
+
+        <v-col>
+          <v-form
+            id="completing-party-form"
+            ref="completingPartyForm"
+            v-model="completingPartyFormValid"
+            v-on:submit.prevent
+          >
+            <!-- Person's Name -->
+            <label class="sub-header">Person's Name</label>
+            <v-row no-gutters class="pt-6">
+              <v-col>
+                <v-text-field
+                  filled
+                  class="item"
+                  label="First Name"
+                  id="person__first-name"
+                  :rules="firstNameRules"
+                  v-model="completingParty.firstName"
+                />
+              </v-col>
+              <v-col class="px-4">
+                <v-text-field
+                  filled
+                  class="item"
+                  label="Middle Name"
+                  id="person__middle-name"
+                  :rules="middleNameRules"
+                  v-model="completingParty.middleName"
+                />
+              </v-col>
+              <v-col>
+                <v-text-field
+                  filled
+                  class="item"
+                  label="Last Name"
+                  id="person__last-name"
+                  :rules="lastNameRules"
+                  v-model="completingParty.lastName"
+                />
+              </v-col>
+            </v-row>
+
+            <!-- Mailing Address -->
+            <div class="mt-2">
+              <label class="sub-header">Mailing Address</label>
+              <div class="address-wrapper pt-6">
+                <base-address
+                  ref="mailingAddress"
+                  :editing="true"
+                  :schema="addressSchema"
+                  :address="currentCompletingParty && currentCompletingParty.mailingAddress"
+                  @update:address="completingParty.mailingAddress = $event"
+                  @valid="addressValid = $event"
+                />
```

Comment:
I think you missed my dev demo yesterday but I talked about an invalid Base Address component not making the form invalid, and I did this to work around it:

https://github.com/bcgov/business-create-ui/blob/a3c99025d0ceb6ad4831ef88c065e80edfa8e3d6/src/components/Registration/RegAddEditOrgPerson.vue#L232

Reply:
In one case I had to do both: once everything is valid, go back and edit an address field to make the address form invalid. Does the main form become invalid too?

If you just moved the current code here from a UI, don't worry too much about it. (Kaine has said that he'd rather get more stuff implemented than get small stuff perfect, as long as we don't accept bad data.)

Does this component (and address) validate correctly currently? If so, no change needed here.

Reply:
Isolated yes, when the validate prop is passed it will validate both the Name inputs and the Address inputs against the schema provided. 
Obviously, this state needs to be handled at an app level as well before filing (ie ensure there is a completing party before filing) but that can be handled locally.